portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,65 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.Comment;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that creates and reads comments. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+    /**
+    Reads comments.
+    @param request the request message
+    @param response the response message
+    @throws IOException if there is an I/O error
+    */
```

Comment:
Almost! Some spacing and *'s are missing.
```suggestion
    /**
    * Reads comments.
    * @param request the request message
    * @param response the response message
    * @throws IOException if there is an I/O error
    */
```

Reply:
Please also apply to doPost.